wqflask/tests/unit/utility/test_authentication_tools.py

```diff
@@ -38,14 +37,15 @@ class TestCheckResourceAvailability(unittest.TestCase):
     """Test methods related to checking the resource availability"""
     @mock.patch('utility.authentication_tools.add_new_resource')
     @mock.patch('utility.authentication_tools.Redis')
-    @mock.patch('utility.authentication_tools.g', mock.Mock())
+    @mock.patch('utility.authentication_tools.g', TestUserSession())
```

Comment:
**nitpick**: It's best to pass in mock objects as patches instead of passing actual objects since you can make assertions like checking what the method was passed with.

Reply:
I changed this because it was throwing an error when using mock.Mock() at the line that says "str.encode(g.user_session.user_id)" in line 34 of utility/authentication_tools.py

Though now that I think about it, it might be possible to change that line to convert the contents of Redis.smembers("super_users") to strings instead of converting g.user_session.user_id to bytes, which should make the original line in the unit test work.